mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/binding/impl/AbstractBindingContext.java

```diff
@@ -6,20 +6,49 @@
  */
 package org.hibernate.search.mapper.pojo.bridge.binding.impl;
 
+import java.lang.invoke.MethodHandles;
+import java.util.Map;
+import java.util.Optional;
+
 import org.hibernate.search.engine.environment.bean.BeanResolver;
 import org.hibernate.search.mapper.pojo.bridge.binding.BindingContext;
+import org.hibernate.search.mapper.pojo.logging.impl.Log;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
 
 abstract class AbstractBindingContext implements BindingContext {
 
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
 	private final BeanResolver beanResolver;
+	private final Map<String, Object> params;
 
-	AbstractBindingContext(BeanResolver beanResolver) {
+	AbstractBindingContext(BeanResolver beanResolver, Map<String, Object> params) {
 		this.beanResolver = beanResolver;
+		this.params = params;
 	}
 
 	@Override
 	public BeanResolver beanResolver() {
 		return beanResolver;
 	}
 
+	@Override
+	public Object param(String name) {
+		Object value = params.get( name );
+		if ( value == null ) {
+			throw log.paramNotDefined( name );
+		}
+
+		return value;
+	}
+
+	@Override
+	public Optional<Object> optionalParam(String name) {
+		Object value = params.get( name );
+		if ( value == null ) {
+			return Optional.empty();
+		}
+
+		return Optional.of( value );
```

Comment:
Or...

```suggestion
		return Optional.ofNullable( params.get( name ) );
```

Reply:
I never remember that this method exists :P